File: golang/141. Linked List Cycle .go
func hasCycle(head *ListNode) bool {
    curr := head 
    set := make(map[*ListNode]bool)
    
    for curr != nil {
        if ok, _ := set[curr]; ok {
            return true
        }
        set[curr] = true
        curr = curr.Next
    }
    return false
}

File: java/141. Linked List Cycle .java
public class Solution {
    public boolean hasCycle(ListNode head) {

        //we use the hashtable to save 
        Set<ListNode> seen = new HashSet<>();


        while( head != null){

            if(seen.contains(head)) {return true;}
            seen.add(head);

            head = head.next;
        }
        return false;
    }
}
